bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/handler/DigiplexZoneHandler.java

```diff
@@ -51,6 +52,7 @@
 @NonNullByDefault
 public class DigiplexZoneHandler extends BaseThingHandler {
 
+    private @Nullable DigiplexZoneConfiguration config;
```

Comment:
Seems like `zoneNo` can now be stored both directly as a field in a class and inside `DigiplexZoneConfiguration`.
I wonder if we can depend solely on the one from `DigitplexZoneConfiguration` so that entire configuration is stored there?

Reply:
Indeed there are two places to store the same information. The zone (respectively area) number can be stored inside thing properties when using autodiscovery or thing configurations when using configuration files. 

If you decide to use only one (and I think should be thing configuration to be able to use *.things files) then:

-  the autodiscover code from ``DigiplexDiscoveryService`` should be updated to use configurations instead of properties. But I'm not sure exactly if it's ok because I see there is no method ``withConfigurations`` in ``org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder``. I guess you cannot use configuration when are in autodiscovery mode. 
- the code should be kept backward compatible (so properties still needs to be used) because I suppose there are users using this binding with autodiscovered things stored in jasondb. 

So, if you are able to answer to the question: "Is it ok to use configuration during discovery?" then two options are:

 - Yes: the the code can be refactored to use only configuration and try to keep backward compatible;
 - No: use the field ``zoneNo`` in ``DigiplexZoneHandler`` and initialize it from property (when discovery) or from 
configuration (when configuration file)

Reply:
I don't think it's possible to use configuration in `DiscoveryResultBuilder`, but I don't think it blocks us from using `DigiplexZoneConfiguration` to store everything we need. As far as I understand, `getConfigAs(DigiplexZoneConfiguration.class)` will just return empty `DigiplexZoneConfiguration` object in case of autodiscovered zone, which we can then fill with a value retrieved from a property.
So, instead of:
```
if (nodeParm != null) {
    zoneNo = Integer.parseInt(nodeParm);
} else {
    zoneNo = config.zoneNo;
}
```
we will have:
```
if (nodeParm != null) {
    config.zoneNo = Integer.parseInt(nodeParm);
}
```
and we can replace all usages of `zoneNo` with `config.zoneNo`. 
Am I missing something? It's been a while since my last contact with this binding and openhab as such.